### DISCUSSION TITLE
How to get the token balance

### DISCUSSION THREAD

**Discussion by Kamo-Smbatyan** [community/community#143947] [Programming Help] [answered]
### Body

Hey there,
I wanna get the token balance with wallet address in Solana ecosystem
Could anybody share me sample codebase?
Thanks 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Octorcat** [comment]
Hi Kamo
You can use getTokenBalanceAccount
Please refer this code.

const getTokenBalanceByWalletAddress = async (tokenAddress: string): Promise<number> => {
  const tokenMintAddress = new PublicKey(tokenAddress);

  try {
    const tokenAccounts = await conn.getTokenAccountsByOwner(walletPubKey, { mint: tokenMintAddress });

    if (!tokenAccounts.value?.length) {
      return 0;
    }

    const tokenAccount = tokenAccounts.value[0].pubkey;
    const balanceResponse = await conn.getTokenAccountBalance(tokenAccount);
    
    return Number(balanceResponse.value.amount);
  } catch (error) {
    console.error(`Failed to fetch token balance. Error: ${error}`);
    return 0;
  }
}

Have a nice day!

**Reply by Kamo-Smbatyan** [reply, marked as answer]
Awesome.
It works well.
Thanks so much
